### PR TITLE
Update restGetAddressableByAddress to use operator

### LINKED ISSUE #1473: Refactor restGetAddressableByAddress to use new operator pattern
# Refactor Request

### Relevant Package
core-metadata

### Description
Refactor the `restGetAddressableByAddress` API endpoint as per the [Unit Test Refactoring project](https://github.com/edgexfoundry/edgex-go/projects/11). The strategy we are aiming for is the one presented by Trevor and is currently implemented for the `restAddNewDevice` API with the usage of operators. For more details regarding the refactoring strategy please refer to #1452 and #1453 

### Describe the solution you'd like
The `restGetAddressableByAddress` endpoint updated to use the new operator pattern as presented in #1453 

### Describe alternatives you've considered
N/A as this strategy has gone though multiple review cycles and has been iterated on.

### PR DESCRIPTION
**Pull request by AnthonyMBonafide**
Fix #1473

Update the `restGetAddressableByAddress` function to leverage the new
operator pattern which will allow for easier testing.

Add tests to cover the new operator.

Signed-off-by: Anthony M. Bonafide <AnthonyMBonafide@gmail.com>

### CODE REVIEW

**Comment by AnthonyMBonafide** [comment]
After taking a second look, I had my interface names mixed up and had to do a little bit of renaming.

File: internal/core/metadata/operators/addressable/db.go [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// addressable contains functionality for obtaining Addressable data
+package addressable
+
+import contract "github.com/edgexfoundry/go-mod-core-contracts/models"
+
+// AddressLoader provides functionality for obtaining Addressables given an address
+type AddressLoader interface {
+	GetAddressablesByAddress(add string) ([]contract.Addressable, error)
```
Comment (tsconn23) [inline internal/core/metadata/operators/addressable/db.go]: Please spell out `address` for the parameter name

File: internal/core/metadata/operators/addressable/get.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	address  string
+}
+
+// Execute retrieves the addressables from the underlying data-store
+func (n addressLoader) Execute() ([]contract.Addressable, error) {
+	return n.database.GetAddressablesByAddress(n.address)
+}
+
+// NewAddressLoader creates an AddressablesRetreiver
+func NewAddressLoader(db AddressLoader, name string) AddressablesRetreiver {
```
Comment (tsconn23) [inline internal/core/metadata/operators/addressable/get.go]: `name string` should be `address string` I believe.

File: internal/core/metadata/operators/addressable/get.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	return addressLoader{
+		database: db,
+		address:  name}
```
Comment (tsconn23) [inline internal/core/metadata/operators/addressable/get.go]: Fix assignment

File: internal/core/metadata/operators/addressable/get_test.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+			expectedResult: nil,
+			expectedError:  true,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			op := NewAddressLoader(test.mockDb, AddressName)
+			actual, err := op.Execute()
+			if test.expectedError && err == nil {
+				t.Error("Expected an error")
```
Comment (tsconn23) [inline internal/core/metadata/operators/addressable/get_test.go]: `return` after each of these error statements

**Pull request merged**